README.md

```diff
@@ -211,19 +227,41 @@ Clients can be given a name.
 A name is a logical identifier which can be used to associate clients with a particular provider.
 If a name has no associated provider, the global provider is used.
 
+#### Asynchronous Named Provider Configuration   
+
+To register providers asynchronously, ensuring they are initialized before further actions are taken, you can use the `setProviderAndWait` method as shown below:   
+
 ```java
 FeatureProvider scopedProvider = new MyProvider();
 
-// registering the default provider
+// registering the default provider asynchronously
+OpenFeatureAPI.getInstance().setProviderAndWait(LocalProvider());
+// registering a named provider asynchronously
+OpenFeatureAPI.getInstance().setProviderAndWait("clientForCache", new CachedProvider());
+
+// a client backed by the default provider
+Client clientDefault = OpenFeatureAPI.getInstance().getClient();
+// a client backed by CachedProvider
+Client clientNamed = OpenFeatureAPI.getInstance().getClient("clientForCache");
+```   
+   
+#### Synchronous Named Provider Configuration   
+
+For synchronous provider registration, where you set the provider without waiting for initialization, you can use the `setProvider` method directly, as shown in the following example:  
+
+```java
+FeatureProvider scopedProvider = new MyProvider();
+
+// registering the default provider synchronously
```

Comment:
I don't think you need to repeat yourself here. Just mention that named providers can be set in a blocking/non-blocking way, and then link to the previous doc you wrote above.